packages/entity-server/src/routes/hierarchy/types.ts

```diff
@@ -48,24 +56,36 @@ export type FlattenedEntity = FlattableEntityFields[keyof FlattableEntityFields]
 
 export type EntityResponse = EntityResponseObject | FlattenedEntity;
 
-export interface HierarchyContext {
-  entity: EntityType;
+export type CommonContext = {
   hierarchyId: string;
   allowedCountries: string[];
   fields: (keyof ExtendedEntityFields)[];
   filter: EntityFilter;
   field?: keyof FlattableEntityFields;
+};
+
+export interface SingleEntityContext extends CommonContext {
+  entity: EntityType;
+}
+
+export interface MultiEntityContext extends CommonContext {
+  entities: EntityType[];
 }
 
-export interface HierarchyRequest<
-  P = HierarchyRequestParams,
+export interface SingleEntityRequest<
```

Comment:
Bulk renamed HierarchyRequest => SingleEntityRequest

Sorry for all the deltas...